src/app/interseptors/token.interceptor.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import {
+    HttpRequest,
+    HttpHandler,
+    HttpEvent,
+    HttpInterceptor
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class TokenInterceptor implements HttpInterceptor {
+    intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+        if (!request.url.includes('http://localhost:3004/auth/login')) {
+            request = request.clone({
+                setHeaders: {
+                    Authorization: JSON.parse(localStorage.getItem('token')).token
```

Comment:
It would much better to get the authorization token from `UserDataService.getUserData()`